### PR TITLE
Fix a bug for SD35 control net training and improve control net block index

### PR DESCRIPTION
**Pull request by linjiapro**
# What does this PR do?

In SD3.5, ControlNet uses a single-layer transformer and does not rely on encoder_hidden_states. The code has been updated accordingly to prevent crashes.

For SD3.5 Large, which consists of 38 transformer layers, setting the ControlNet layers to 16 layers caused inefficiencies during inference. This was due to the old index logic calculating interval_control as an integer 3, which resulted in the 38-layer SD3.5 Large transformer using only up to layer 13 of the 16-layer ControlNet. To address this, the indexing logic has been revised to calculate interval_control as a floating-point number, ensuring all 16 layers of ControlNet are utilized. This update is backward-compatible, meaning no changes occur for a 19-layer ControlNet.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by linjiapro** [comment]
cc @yiyixuxu @sayakpaul

File: src/diffusers/models/transformers/transformer_sd3.py [hunk 1]
```diff
@@ -424,8 +424,7 @@ def custom_forward(*inputs):
             # controlnet residual
             if block_controlnet_hidden_states is not None and block.context_pre_only is False:
                 interval_control = len(self.transformer_blocks) / len(block_controlnet_hidden_states)
-                interval_control = int(np.ceil(interval_control))
-                hidden_states = hidden_states + block_controlnet_hidden_states[index_block // interval_control]
+                hidden_states = hidden_states + block_controlnet_hidden_states[int(index_block / interval_control)]
```
Comment (linjiapro) [inline src/diffusers/models/transformers/transformer_sd3.py:427]: Please note that this is a change that is backward compatible. 

When len(self.transformer_blocks) is divisible by len(block_controlnet_hidden_states), which is 100% cases for all shipped controlnets, the logic does not change anything.

**Comment by sayakpaul** [comment]
Thanks for the PR. Could you also add some tests to the PR?

File: src/diffusers/models/transformers/transformer_sd3.py (hunk 1 above)
Comment (yiyixuxu) [reply src/diffusers/models/transformers/transformer_sd3.py:427]: ok! but why do we have to change then?

**Comment by yiyixuxu** [comment]
thanks!
can you `make style`?

**Comment by linjiapro** [comment]
@yiyixuxu make style is done!

**Review by yiyixuxu — APPROVED** [review]

**Comment by yiyixuxu** [comment]
thank you @linjiapro !

**Pull request merged**